gnovm/pkg/gnolang/op_exec.go

```diff
@@ -627,7 +627,8 @@ EXEC_SWITCH:
 					if cs.Label != "" && cs.Label != fr.Label {
 						m.PopFrame()
 					} else {
-						m.PopFrameAndReset()
+						bn := findFirstBreakableBlockNode(m.LastBlock().Source, cs.Label)
+						m.PopFrameAndReset2(bn)
```

Comment:
This isn't the right place to solve it. The problem is that GOTO wasn't popping frames that it should.

See https://github.com/gnolang/gno/issues/4098#issuecomment-2833465473